### DISCUSSION TITLE
ParamSpecTree cycles - dependent and independent variables

### DISCUSSION THREAD

**Discussion by ttshaw1** [microsoft/Qcodes#4051] [Q&A]
I'm running an experiment with 3 variables, which I'll call X, Y, and Z. What I want to do is sweep X, measure some data, and calculate Y at each X. So X is a parameter with no setpoints, and Y has X as its setpoint. 

The next thing I want to do is, at each X, calculate Z. I want to be able to plot Z vs. X, but I also want to be able to plot Z vs. Y. So my ideal would be to have Z have X and Y as setpoints, allowing me to easily show either graph in Plottr. As I understand it, this causes an error, as Y is a dependent variable, and so no variable can depend on it: `ParamSpecTree can not have cycles`

Is this the desired behavior? What's the workaround?

**Comment by ttshaw1** [comment]
My current fix is to create a new parameter "Y_indep" and have Z's setpoints be X and Y_indep, but I'm wondering if there's a cleaner way.

**Reply by sohailc** [reply]
Hello, 

Some clarification questions; are you calculating Y and Z with the measured data? Or only with the set point X? Or both? In other words, is the following true: 

Y = f(X, data)
Z = g(X, Y, data)
